pkg/services/secrets/kvstore/kvstore.go

```diff
@@ -35,21 +37,25 @@ func ProvideService(
 			cache: make(map[int64]cachedDecrypted),
 		},
 	}
-	if remoteCheck.ShouldUseRemoteSecretsPlugin() {
+	err := shouldUseRemoteSecretsPlugin(pluginsManager, cfg)
```

Comment:
I think using err we can return both a reason and a checkable condition variable, simplifying things, what do you think? Maybe renaming the function would make it seem less odd?

Reply:
That's true, though it feels like a misuse of error to me, since we debug log the error output. Maybe it is just my Java background speaking!

If we rename it though, it could just be `evaluateUseSecretsPlugin()` and I wouldn't feel like it should return a bool. Your call though, I will approve either way 👍

Reply:
I also feel it is weird to use an error like that (also Java background :) ). 
 
As far as I can see, the returned error on `shouldUseRemoteSecretsPlugin` is only logged, but not returned outside the function or read to see the kind of error. 
What if we change the return value of `shouldUseRemoteSecretsPlugin` to a boolean, and log the error or information needed within that function?
I could also help clean the "if {} else {} " block below which has the "happy path" within the else

Reply:
The plan was to remove the logger within the function so we don't need a global logger or an argument logger just to serve that purpose of logging the reason for debug purposes.

Reply:
Oh right, sorry I circle back to that issue